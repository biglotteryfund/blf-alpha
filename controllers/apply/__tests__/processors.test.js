/* eslint-env jest */
'use strict';
const fs = require('fs');
const path = require('path');
const yaml = require('js-yaml');
const nodemailer = require('nodemailer');
const { get } = require('lodash');

const { flattenFormData, stepsWithValues } = require('../helpers');
const langData = fs.readFileSync(path.resolve(__dirname, '../../../config/locales/en.yml'), 'utf-8');
const enCopy = yaml.safeLoad(langData);

function cleanMailForSnaphot(info) {
    info.message = info.message
        .toString()
        .replace(/Message-ID: .*\n/, '')
        .replace(/Date: .*\n/, '')
        .replace(/ boundary=.*\n/, '')
        .replace(/----.*\n/gm, '');
    return info;
}

const mockTransport = nodemailer.createTransport({
    streamTransport: true,
    newline: 'unix',
    buffer: true
});

describe('Reaching communities', () => {
    const form = require('../reaching-communities/form-model');
    const formCopy = get(enCopy, form.lang);
    const processor = require('../reaching-communities/processor');

    const mockFormData = {
        'step-1': {
            'your-idea': 'Test idea'
        },
        'step-2': {
            location: 'South West',
            'project-location': 'London'
        },
        'step-3': {
            'organisation-name': 'Test Organisation',
            'additional-organisations': ''
        },
        'step-4': {
            'first-name': 'Example',
            'last-name': 'Person',
            email: 'example@example.com',
            'phone-number': '03454102030'
        }
    };

    it('should create success emails', async () => {
        const results = await processor(
            {
                data: flattenFormData(mockFormData),
                copy: formCopy,
                stepsWithValues: stepsWithValues(form.steps, mockFormData)
            },
            mockTransport
        );

        const [customerEmail, internalEmail] = results.map(cleanMailForSnaphot);

        expect(customerEmail.message).toMatchSnapshot();
        expect(internalEmail.message).toMatchSnapshot();
    });
});

describe('Digital fund', () => {
    const form = require('../digital-fund/form-model').strand1;
    const formCopy = get(enCopy, form.lang);
    const processor = require('../digital-fund/processor');

    const mockFormData = {
        'step-1': {
            name: 'Example Person',
            email: 'example@example.com',
            'organisation-name': 'Test Organisation',
            'about-your-organisation': 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro, inventore.'
        }
    };

    it('should create success emails', async () => {
        const results = await processor(
            {
                form: form,
                locale: 'en',
                data: flattenFormData(mockFormData),
                stepsWithValues: stepsWithValues(form.steps, mockFormData),
                copy: formCopy
            },
            mockTransport
        );

        const [customerEmail, internalEmail] = results.map(cleanMailForSnaphot);
        expect(customerEmail.message).toMatchSnapshot();
        expect(internalEmail.message).toMatchSnapshot();
    });
});

describe('Youth capacity fund', () => {
    const form = require('../youth-capacity/form-model');
    const formCopy = get(enCopy, form.lang);
    const { processor, formatDataForStorage } = require('../youth-capacity/processor');

    const mockFormData = {
        'step-1': { 'current-work': 'Test' },
        'step-2': { location: 'Barking and Dagenham', money: 'Test' },
        'step-3': { 'project-aims': 'Develop partnerships', 'people-and-communities': 'Test' },
        'step-4': {
            'organisation-name': 'Test organisation',
            'contact-name': 'Example',
            'contact-email': 'example@example.com',
            'contact-phone': '3454102030'
        }
    };

    it('should format data for storage', () => {
        const stepsData = stepsWithValues(form.steps, mockFormData);
        const result = formatDataForStorage(stepsData, formCopy);
        expect(result).toMatchSnapshot();
    });

    it('should create success emails', async () => {
        const results = await processor(
            {
                form: form,
                data: flattenFormData(mockFormData),
                copy: get(enCopy, form.lang),
                stepsWithValues: stepsWithValues(form.steps, mockFormData)
            },
            mockTransport,
            false
        );

        const [customerEmail, internalEmail] = results.map(cleanMailForSnaphot);

        expect(customerEmail.message).toMatchSnapshot();
        expect(internalEmail.message).toMatchSnapshot();
    });
});
