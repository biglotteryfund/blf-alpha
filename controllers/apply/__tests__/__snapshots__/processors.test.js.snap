// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Digital fund should create success emails 1`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: Thank you for getting in touch with the Big Lottery Fund!
MIME-Version: 1.0

Content-Type: text/plain
Content-Transfer-Encoding: quoted-printable

 Thank you for getting in touch.

Here is a copy of the information you've =
submitted. We'll be in touch as soon as we can.

YOUR DETAILS
YOUR NAME
Example Person

EMAIL ADDRESS
example@example.com

ORGANISATION NAME
Test Organisation

DATA PROTECTION
The Big Lottery Fund is a public body =
with a duty to distribute National Lottery and other money in grants for =
good causes. We use
the personal data you provide, such as contact details =
for individuals at your organisation, to help you apply for a grant and to
assess your application. We may share your personal data with organisations=
 which help us with our grant making activities or
others which have a =
legitimate interest in our work or have funded your grant. We will only =
share personal data which they need to
carry out their work and subject to =
appropriate safety measures. Our Data Protection and Privacy Notice gives =
more information
about how we store and use personal data and the lawful =
basis for this. Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection [https://www.=
biglotteryfund.org.uk/data-protection] or contact us to request
a hard copy. The Notice may be updated from time to time.
Content-Type: text/html
Content-Transfer-Encoding: quoted-printable

<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"padding: 20px;=
 max-width: 75%; margin: auto; color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">
        <div class=3D\\"logo\\" style=3D\\"margin-bottom: 20px; text-align: =
right; display: block;\\">
            <img src=3D\\"https://www.biglotteryfund=
.org.uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

       =20
   =20

    <p style=3D\\"color: #333; font-family: arial; font-weight: 400; =
font-size: 20px; line-height: 1.35;\\">Thank you for getting in touch.</p>
    <p style=3D\\"color: #333; font-family: arial; font-weight: 400; =
font-size: 20px; line-height: 1.35;\\">Here is a copy of the information =
you've submitted. We'll be in touch as soon as we can.</p>

   =20
    <div class=3D\\"summary\\">
       =20
            <h4 style=3D\\"margin: 0 0=
 20px; color: #e20087; font-size: 16px; border-bottom: 1px solid #e20087; =
padding-bottom: 3px; font-weight: 700;\\">Your details</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Your name</h5>
                       =20
                           =
 <p style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">Example Person</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Email address</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">example@example.com</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Organisation name</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Test Organisation</p>
                       =20
                   =20
                   =20
                   =20
       =20
    </div>



       =20

        <h5 class=3D\\"space-above\\" =
style=3D\\"font-weight: 500; margin: 0 0 20px; color: #e20087; font-size: =
14px; margin-top: 50px;\\">Data protection</h5>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">The Big Lottery Fund is a public body with a duty to distribute =
National Lottery and other money in grants for good causes. We use the =
personal data you provide, such as contact details for individuals at your =
organisation, to help you apply for a grant and to assess your application.=
</small>
        <small style=3D\\"display: block; font-size: 11px; color: =
#7a7a7a; margin-bottom: 10px;\\">We may share your personal data with =
organisations which help us with our grant making activities or others =
which have a legitimate interest in our work or have funded your grant. We =
will only share personal data which they need to carry out their work and =
subject to appropriate safety measures.</small>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">Our Data Protection and Privacy Notice gives more information about =
how we store and use personal data and the lawful basis for this. Please =
read the full Notice which is published on our website at <a =
href=3D\\"https://www.biglotteryfund.org.uk/data-protection\\" style=3D\\"color: =
#365cc9;\\">www.biglotteryfund.org.uk/data-protection</a> or contact us to =
request a hard copy. The Notice may be updated from time to time.</small>
    </body>
</html>

"
`;

exports[`Digital fund should create success emails 2`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: New submission for Digital Fund Strand 1 from Test Organisation
MIME-Version: 1.0

Content-Type: text/plain
Content-Transfer-Encoding: quoted-printable

 NEW SUBMISSION FOR DIGITAL FUND STRAND 1
YOUR DETAILS
YOUR NAME
Example Person

EMAIL ADDRESS
example@example.com

ORGANISATION NAME
Test Organisation

DATA PROTECTION
The Big Lottery Fund is a public body =
with a duty to distribute National Lottery and other money in grants for =
good causes. We use
the personal data you provide, such as contact details =
for individuals at your organisation, to help you apply for a grant and to
assess your application. We may share your personal data with organisations=
 which help us with our grant making activities or
others which have a =
legitimate interest in our work or have funded your grant. We will only =
share personal data which they need to
carry out their work and subject to =
appropriate safety measures. Our Data Protection and Privacy Notice gives =
more information
about how we store and use personal data and the lawful =
basis for this. Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection [https://www.=
biglotteryfund.org.uk/data-protection] or contact us to request
a hard copy. The Notice may be updated from time to time.
Content-Type: text/html
Content-Transfer-Encoding: quoted-printable

<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"padding: 20px;=
 max-width: 75%; margin: auto; color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">
        <div class=3D\\"logo\\" style=3D\\"margin-bottom: 20px; text-align: =
right; display: block;\\">
            <img src=3D\\"https://www.biglotteryfund=
.org.uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

       =20
    <h1 style=3D\\"font-weight: 500; margin: 0 0 20px; color: #e20087; =
font-size: 26px;\\">New submission for Digital Fund Strand 1</h1>
   =20
    <div class=3D\\"summary\\">
       =20
            <h4 style=3D\\"margin: 0 0=
 20px; color: #e20087; font-size: 16px; border-bottom: 1px solid #e20087; =
padding-bottom: 3px; font-weight: 700;\\">Your details</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Your name</h5>
                       =20
                           =
 <p style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">Example Person</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Email address</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">example@example.com</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Organisation name</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Test Organisation</p>
                       =20
                   =20
                   =20
                   =20
       =20
    </div>



       =20

        <h5 class=3D\\"space-above\\" =
style=3D\\"font-weight: 500; margin: 0 0 20px; color: #e20087; font-size: =
14px; margin-top: 50px;\\">Data protection</h5>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">The Big Lottery Fund is a public body with a duty to distribute =
National Lottery and other money in grants for good causes. We use the =
personal data you provide, such as contact details for individuals at your =
organisation, to help you apply for a grant and to assess your application.=
</small>
        <small style=3D\\"display: block; font-size: 11px; color: =
#7a7a7a; margin-bottom: 10px;\\">We may share your personal data with =
organisations which help us with our grant making activities or others =
which have a legitimate interest in our work or have funded your grant. We =
will only share personal data which they need to carry out their work and =
subject to appropriate safety measures.</small>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">Our Data Protection and Privacy Notice gives more information about =
how we store and use personal data and the lawful basis for this. Please =
read the full Notice which is published on our website at <a =
href=3D\\"https://www.biglotteryfund.org.uk/data-protection\\" style=3D\\"color: =
#365cc9;\\">www.biglotteryfund.org.uk/data-protection</a> or contact us to =
request a hard copy. The Notice may be updated from time to time.</small>
    </body>
</html>

"
`;

exports[`Reaching communities should create success emails 1`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: Thank you for getting in touch with the Big Lottery Fund!
MIME-Version: 1.0

Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable

 Thank you for getting in touch with us.

You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in
touch within fifteen working days to let =
you know if we can help.

In the meantime, you can find out more about our =
funding and how we can support your community on our website:=20
www.biglotteryfund.org.uk/england [https://www.biglotteryfund.org.=
uk/england]

Thanks,

The Big Lottery Fund

YOUR INFORMATION
YOUR IDEA
BRIEFLY EXPLAIN YOUR IDEA AND WHY IT=E2=80=99LL MAKE A DIFFERENCE
Test idea

PROJECT LOCATION
SELECT ALL REGIONS THAT APPLY
South West

PROJECT LOCATION
London

YOUR ORGANISATION
LEGAL NAME
Test Organisation

YOUR DETAILS
FIRST NAME
Example

LAST NAME
Person

EMAIL ADDRESS
example@example.com

PHONE NUMBER
03454102030

DATA PROTECTION
The Big Lottery Fund is a public body with a duty to distribute National =
Lottery and other money in grants for good causes. We use
the personal data you provide, such as contact details for individuals at =
your organisation, to help you apply for a grant and to
assess your application. We may share your personal data with organisations=
 which help us with our grant making activities or
others which have a =
legitimate interest in our work or have funded your grant. We will only =
share personal data which they need to
carry out their work and subject to =
appropriate safety measures. Our Data Protection and Privacy Notice gives =
more information
about how we store and use personal data and the lawful =
basis for this. Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection or contact us to =
request a hard copy. The Notice may be updated from time to
time.
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable


<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"padding: 20px;=
 max-width: 75%; margin: auto; color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">
        <div class=3D\\"logo\\" style=3D\\"margin-bottom: 20px; text-align: =
right; display: block;\\">
            <img src=3D\\"https://www.biglotteryfund=
.org.uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

        <p style=3D\\"color: #333; font-family: arial; font-weight: 400; =
font-size: 20px; line-height: 1.35;\\">Thank you for getting in touch with us=
.</p>

        <p style=3D\\"color: #333; font-family: arial; font-weight: =
400; font-size: 20px; line-height: 1.35;\\">You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in touch within fifteen working days to let =
you know if we can help.</p>
        <p style=3D\\"color: #333; font-family: =
arial; font-weight: 400; font-size: 20px; line-height: 1.35;\\">In the =
meantime, you can find out more about our funding and how we can support =
your community on our website: <a href=3D\\"https://www.biglotteryfund.org.=
uk/england\\" style=3D\\"color: #365cc9;\\">www.biglotteryfund.org.=
uk/england</a></p>
        <p style=3D\\"color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">Thanks,</p>
        <p style=3D\\"color: #333; font-family: arial; font-weight: 400; =
font-size: 20px; line-height: 1.35;\\"><strong>The Big Lottery =
Fund</strong></p>

        <h2 class=3D\\"space-above\\" style=3D\\"font-weight: =
500; margin: 0 0 20px; color: #e20087; font-size: 22px; margin-top: =
50px;\\">Your information</h2>

       =20
    <div class=3D\\"summary\\">
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; =
font-weight: 700;\\">Your idea</h4>
           =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 14px; font-weight: 700;\\">Briefly explain =
your idea and why it=E2=80=99ll make a difference</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Test idea</p>
                       =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; =
font-weight: 700;\\">Project location</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Select all regions that apply</h5>
                       =20
                            <p style=3D\\"color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">South West</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 14px; font-weight: 700;\\">Project =
location</h5>
                       =20
                            <p =
style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">London</p>
                       =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 16px; border-bottom: 1px solid #e20087; =
padding-bottom: 3px; font-weight: 700;\\">Your organisation</h4>
           =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Legal name</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Test Organisation</p>
                       =20
                   =20
                   =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; =
font-weight: 700;\\">Your details</h4>
           =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 14px; font-weight: 700;\\">First name</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Example</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Last name</h5>
                       =20
                           =
 <p style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">Person</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Email address</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">example@example.com</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Phone number</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">03454102030</p>
                       =20
                   =20
       =20
    </div>


        <h5 class=3D\\"space-above\\" =
style=3D\\"font-weight: 500; margin: 0 0 20px; color: #e20087; font-size: =
14px; margin-top: 50px;\\">Data protection</h5>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">The Big Lottery Fund is a public body with a duty to distribute =
National Lottery and other money in grants for good causes. We use the =
personal data you provide, such as contact details for individuals at your =
organisation, to help you apply for a grant and to assess your application.=
</small>
        <small style=3D\\"display: block; font-size: 11px; color: =
#7a7a7a; margin-bottom: 10px;\\">We may share your personal data with =
organisations which help us with our grant making activities or others =
which have a legitimate interest in our work or have funded your grant. We =
will only share personal data which they need to carry out their work and =
subject to appropriate safety measures.</small>
        <small =
style=3D\\"display: block; font-size: 11px; color: #7a7a7a; margin-bottom: =
10px;\\">Our Data Protection and Privacy Notice gives more information about =
how we store and use personal data and the lawful basis for this. Please =
read the full Notice which is published on our website at www.=
biglotteryfund.org.uk/data-protection or contact us to request a hard copy.=
 The Notice may be updated from time to time.</small>
    </body>
</html>

"
`;

exports[`Reaching communities should create success emails 2`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: New idea submission from website: Test Organisation
MIME-Version: 1.0

Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable

 A CUSTOMER HAS SUBMITTED AN IDEA FROM THE REACHING COMMUNITIES & =
PARTNERSHIPS WEB FORM.
The details they provided are included below:

CUSTOMER INFORMATION
YOUR DETAILS
FIRST NAME
Example

LAST NAME
Person

EMAIL ADDRESS
example@example.com

PHONE NUMBER
03454102030

YOUR ORGANISATION
LEGAL NAME
Test Organisation

PROJECT LOCATION
SELECT ALL REGIONS THAT APPLY
South West

PROJECT LOCATION
London

YOUR IDEA
BRIEFLY EXPLAIN YOUR IDEA AND WHY IT=E2=80=99LL MAKE A DIFFERENCE
Test idea
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable


<!doctype html>
<html>
    <head>
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), local('Poppins-Light'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLDz8Z1xlEA.ttf) =
format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), =
local('Poppins-Medium'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8=
kv8JHgFVrLGT9Z1xlEA.ttf) format('truetype');}
</style>
        <meta charset=3D\\"UTF-8\\">
    </head>
    <body style=3D\\"padding: =
20px; max-width: 75%; margin: auto; color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">
        <h1 style=3D\\"font-weight: 500; margin: 0 0 20px; color: #e20087; =
font-size: 26px;\\">A customer has submitted an idea from the Reaching =
Communities &amp; Partnerships web form.</h1>
        <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">The details they provided are included below:</p>
        <h2 class=3D\\"space-above\\" style=3D\\"font-weight: 500; margin: 0 0 =
20px; color: #e20087; font-size: 22px; margin-top: 50px;\\">Customer =
information</h2>
       =20
    <div class=3D\\"summary\\">
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 16px;=
 border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your details</h4>
           =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 14px; font-weight: 700;\\">First name</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">Example</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Last name</h5>
                       =20
                           =
 <p style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">Person</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Email address</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">example@example.com</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 14px; font-weight: 700;\\">Phone number</h5>
                       =20
                            <p style=3D\\"color: =
#333; font-family: arial; font-weight: 400; font-size: 20px; line-height: 1=
.35;\\">03454102030</p>
                       =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; =
font-weight: 700;\\">Your organisation</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Legal name</h5>
                       =20
                          =
  <p style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size:=
 20px; line-height: 1.35;\\">Test Organisation</p>
                       =20
                   =20
                   =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 20px; color: #e20087; =
font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; =
font-weight: 700;\\">Project location</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Select all regions that apply</h5>
                       =20
                            <p style=3D\\"color: #333; font-family: arial; =
font-weight: 400; font-size: 20px; line-height: 1.35;\\">South West</p>
                       =20
                   =20
                   =20
                   =20
                        <h5 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 14px; font-weight: 700;\\">Project =
location</h5>
                       =20
                            <p =
style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">London</p>
                       =20
                   =20
       =20
            <h4 style=3D\\"margin: 0 0 =
20px; color: #e20087; font-size: 16px; border-bottom: 1px solid #e20087; =
padding-bottom: 3px; font-weight: 700;\\">Your idea</h4>
           =20
                   =20
                   =20
                        <h5 =
style=3D\\"margin: 0 0 20px; color: #e20087; font-size: 14px; font-weight: =
700;\\">Briefly explain your idea and why it=E2=80=99ll make a =
difference</h5>
                       =20
                            <p =
style=3D\\"color: #333; font-family: arial; font-weight: 400; font-size: =
20px; line-height: 1.35;\\">Test idea</p>
                       =20
                   =20
       =20
    </div>

    </body>
</html>

"
`;
